scripts/scil_tractogram_filter_collisions.py

```diff
@@ -0,0 +1,174 @@
+#! /usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Given an input tractogram and a text file containing a diameter for each
+streamline, filters all intersecting ones and saves the resulting
+tractogram and diameters.
```

Comment:
Is this script deterministic? i.e. you run it twice from same input, you get the same output?
If you shuffle streamlines in the input, do you get same output?
The colliding streamlines get filtered out? 

Can you be a bit more detailed in your small summary please